src/application/service/CreateTest.ts

```diff
@@ -2,32 +2,64 @@ import { UserId } from '../../domain/model/user/UserId';
 import { Test } from '../../domain/model/test/Test';
 import { TestId } from '../../domain/model/test/TestId';
 import { TestTypeId } from '../../domain/model/testType/TestTypeId';
-import { Results } from '../../domain/model/test/Results';
-
 import { TestRepository } from '../../domain/model/test/TestRepository';
 import { TestTypeRepository } from '../../domain/model/testType/TestTypeRepository';
-
-import { getTestTypes } from './index';
-import { TestCommand } from '../../api/interface/index';
-
+import { TestCommand, TestResultsCommand } from '../../api/interface';
 import { testResultsFactory } from './index';
-
 import { DomainValidationError } from '../../domain/model/DomainValidationError';
 import { ResourceNotFoundError } from '../../domain/model/ResourceNotFoundError';
+import { TestType } from '../../domain/model/testType/TestType';
+import { User } from '../../domain/model/user/User';
+import { AccessDeniedError } from '../../domain/model/AccessDeniedError';
 
-export class CreateOrUpdateTest {
+export class CreateTest {
   constructor(
     private testRepository: TestRepository,
     private testTypeRepository: TestTypeRepository
   ) {}
 
   public async execute(
-    actorUserId: string,
+    actor: User,
```

Comment:
Sig is a bit odd, two users but one is passed as an object, one as an id?  I have been a bit confused over which approach is preferred on these.

Reply:
Good point. I initially also kept it with merely passing the actorId but then I realised that the actor will always be present on the presentation and we are querying the user repo twice. So for efficiency's shake I opted to send down/and make the application service require an instance of actor.